src/custom/utils/signatures.ts

```diff
@@ -137,9 +139,19 @@ async function _signPayload(
   payload: any,
   signFn: typeof _signOrder | typeof _signOrderCancellation,
   signer: Signer,
-  signingMethod: 'default' | 'v4' | 'int_v4' | 'v3' | 'eth_sign' = 'v4'
+  signingMethod: 'default' | 'v4' | 'int_v4' | 'v3' | 'eth_sign' = 'v4',
+  isSmartContract: boolean
 ): Promise<SigningResult> {
-  const signingScheme = signingMethod === 'eth_sign' ? SigningScheme.ETHSIGN : SigningScheme.EIP712
+  let signingScheme: SigningScheme
+  if (isSmartContract) {
```

Comment:
Why passing the `signingMethod` instead of specifying the `signingMethod` as `eip1271`?

Reply:
Not sure what you're referring to, but if it's why I left `signingMethod`,
As I did not want to change the original structure, (taking in consideration that this method calls itself in case of fallbacks), I decided to keep it that way